lazy-imports-import.html

```diff
@@ -22,23 +23,26 @@
       var imports = fromElement.querySelectorAll(query);
       var promises = [];
       for (var i = 0; i < imports.length; i++) {
-        promises.push(new Promise(function(resolve) {
-          var href = imports[i].getAttribute('href');
-          var onLoad = function() {
-            resolve({loaded: href});
-          };
-          var onFailure = function() {
-            resolve({failed: href});
-          };
-          var resolved = callingElement.resolveUrl(href);
-          if (!__loadedImports[resolved]) {
-            __loadedImports[resolved] = true;
+        var href = imports[i].getAttribute('href');
+        var resolvedUrl = callingElement.resolveUrl(href);
+
+        if (__failedImports[resolvedUrl] || !__loadedImports[resolvedUrl]) {
```

Comment:
I agree 100%, that it could be simpler by ignoring the failedImports case below. In fact, I had that originally in fact.

So the case that I was trying to solve with maintaining the failed imports was to be able to try reloading failed imports. 

In the current state of the code, failedImports are cached and will never attempt re-fetching them if called again. This could be useful if an app attempted to enhance the experience using lazy-imports but was offline and just came back online. 

Maybe it would be good to have a flag on `importLazyGroup` for skipping the cache of failed? Or maybe this isn't a use-case that we want to cover. I'm open to reverting the failedImports handling but think we should rename the variable holding the cache from "loaded" to something that indicates they were just attempted.